### PR TITLE
Add cfg fun

### PR DESCRIPTION
**Pull request by op-hunter**
**What type of PR is this?**

 feature

**What this PR does / why we need it:**
add  wal config set functions and gtests for wal config

**Which issue(s) this PR fixes:**
#1510

**Special notes for your reviewer:**
no

**Does this PR introduce a user-facing change?:**
no

**Additional documentation (e.g. design docs, usage docs, etc.):**

### CODE REVIEW

**Comment by JinHai-CN** [comment]
CHANGELOG need to update and which issue this PR belongs to?

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ Please mark all change in change log and use the issue from GitHub
 -   \#1298 Unittest failed when on CPU2GPU case
 -   \#1359 Negative distance value returned when searching with HNSW index type
 -   \#1429 Server crashed when searching vectors using GPU
+-   \#1510 Add set interfaces for WAL configurations
```
Comment (cydrain) [inline CHANGELOG.md]: please keep order

File: core/src/server/Config.cpp [hunk 2]
```diff
@@ -74,6 +74,20 @@ Config::Config() {
 
     std::string node_gpu_build_res = std::string(CONFIG_GPU_RESOURCE) + "." + CONFIG_GPU_RESOURCE_BUILD_INDEX_RESOURCES;
     config_callback_[node_gpu_build_res] = empty_map;
+
```
Comment (cydrain) [inline core/src/server/Config.cpp]: WAL config need not add these callbacks

File: core/src/server/Config.cpp [hunk 3]
```diff
@@ -402,6 +422,16 @@ Config::SetConfigCli(const std::string& parent_key, const std::string& child_key
 #endif
     } else if (parent_key == CONFIG_TRACING) {
         return Status(SERVER_UNSUPPORTED_ERROR, "Not support set tracing_config currently");
+    } else if (CONFIG_WAL == parent_key) {
```
Comment (cydrain) [inline core/src/server/Config.cpp]: keep the coding style as other lines, use "child_key == xxx"

File: core/src/server/Config.cpp [hunk 4]
```diff
@@ -1688,6 +1737,7 @@ Config::GetWalConfigBufferSize(int64_t& buffer_size) {
 Status
 Config::GetWalConfigWalPath(std::string& wal_path) {
     wal_path = GetConfigStr(CONFIG_WAL, CONFIG_WAL_WAL_PATH, CONFIG_WAL_WAL_PATH_DEFAULT);
+    CONFIG_CHECK(CheckWalConfigWalPath(wal_path));
```
Comment (cydrain) [inline core/src/server/Config.cpp]: need not check here

**Pull request merged**